tools/js-sdk-release-tools/src/common/utils.ts

```diff
@@ -385,9 +392,34 @@ export async function cleanUpPackageDirectory(
     packageDirectory: string,
     runMode: RunMode,
 ): Promise<void> {
-    // Preserve test directory and assets.json file in non-SpecPullRequest mode
-    const shouldPreserveTestAndAssets = runMode !== RunMode.SpecPullRequest;
+    // Preserve test directory and assets.json file only in Release and Local modes
+    // In SpecPullRequest and Batch modes, remove everything
+    const shouldPreserveTestAndAssets = runMode !== RunMode.SpecPullRequest && runMode !== RunMode.Batch;
     const entriesToPreserve = shouldPreserveTestAndAssets ? ["test", "assets.json"] : [];
 
     await cleanUpDirectory(packageDirectory, entriesToPreserve);
+}
+
+export async function getPackageNameFromTspConfig(typeSpecDirectory: string): Promise<string | undefined> {
+
+    const tspConfig = await resolveOptions(typeSpecDirectory);
+    const emitterOptions = tspConfig.options?.[emitterName];
+    
+    // First try to get from package-details.name which is the actual NPM package name
+    if (emitterOptions?.['package-details']?.name) {
```

Comment:
if package name is not provide in `tspconfig.yaml`, can js emitter still generate sdk? @MaryGao